sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -150,6 +150,62 @@ def test_incr_mean_variance_axis():
                 assert_array_equal(X.shape[axis], n_incr)
 
 
+@pytest.mark.parametrize(
+    "X1, X2",
+    [
+        (sp.random(5, 2, density=0.8, random_state=0).tocsr(),
+         sp.random(13, 2, density=0.8, random_state=0).tocsr()),
+        (sp.random(5, 2, density=0.8, random_state=0).tocsr(),
+         sp.hstack([sp.csr_matrix(np.full((13, 1), fill_value=np.nan)),
+                    sp.random(13, 1, density=0.8, random_state=42)],
+                   format="csr"))
+    ]
+)
+def test_incr_mean_variance_axis_equivalence_mean_variance(X1, X2):
+    # non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/16448
+    # check that computing the incremental mean and variance is equivalent to
+    # computing the mean and variance on the stacked dataset.
+    axis = 0
+    last_mean, last_var = np.zeros(X1.shape[1]), np.zeros(X1.shape[1])
+    last_n = np.zeros(X1.shape[1], dtype=np.int64)
+    updated_mean, updated_var, updated_n = incr_mean_variance_axis(
+        X1, axis, last_mean, last_var, last_n
+    )
+    updated_mean, updated_var, updated_n = incr_mean_variance_axis(
+        X2, axis, updated_mean, updated_var, updated_n
+    )
+    X = sp.vstack([X1, X2])
+    assert_allclose(updated_mean, np.nanmean(X.A, axis=axis))
+    assert_allclose(updated_var, np.nanvar(X.A, axis=axis))
+    assert_allclose(updated_n, np.count_nonzero(~np.isnan(X.A), axis=0))
+
+
+@pytest.mark.parametrize(
+    "X2",
+    [sp.random(0, 1, density=0.8, random_state=0).tocsr(),
+     sp.csr_matrix(np.full((3, 1), fill_value=np.nan))],
+    ids=["empty", "full nan"]
+)
+def test_incr_mean_variance_no_new_n(X2):
+    # check the behaviour when we update the variance with an empty
```

Comment:
don't you already test that in the test above ?

Reply:
not the case where the sparse matrix is empty (0 rows).
It is difficult to incorporate this case in the above test. Here, we don't look at the equivalence with the stacking of the data.

Reply:
oh right I missed the 0 rows :)
However the second X2 is a duplicate of above, right ?

Reply:
now yest let's remove it.